lux/executor/PandasExecutor.py

```diff
@@ -394,6 +394,18 @@ def execute_2D_binning(vis: Vis) -> None:
             x_attr = vis.get_attr_by_channel("x")[0].attribute
             y_attr = vis.get_attr_by_channel("y")[0].attribute
 
+            if vis.data[x_attr].dtype == np.dtype('O'):
+                try:
+                    vis.data[x_attr] = vis.data[x_attr].astype(float)
+                except ValueError:
+                    pass
+
+            if vis.data[y_attr].dtype == np.dtype('O'):
+                try:
+                    vis.data[y_attr] = vis.data[y_attr].astype(float)
+                except ValueError:
+                    pass
+
```

Comment:
How does this conversion interact with the warning we show [here](https://github.com/lux-org/lux/blob/master/lux/executor/PandasExecutor.py#L257)? For performance reasons, should we be doing this outside, as a more general dataframe sanitization step, instead of within specific execute visualization calls, so that the conversion is only done once per column, instead of every time per visualization?

Reply:
> How does this conversion interact with the warning we show [here](https://github.com/lux-org/lux/blob/master/lux/executor/PandasExecutor.py#L257)?

I don't see that warning about mixed types so it not reaching that path.

> For performance reasons, should we be doing this outside, as a more general dataframe sanitization step, instead of within specific execute visualization calls, so that the conversion is only done once per column, instead of every time per visualization?

I do agree, this type of visualization-specific sanitation happened in a previous issue as well and its not ideal. It seems we need a centralize sanitation mechanism that doesn't exist yet. I can open an issue for this.

Reply:
Sure let's open a separate issue for this.